api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -925,4 +927,62 @@ public void getConceptNameTagByName_shouldReturnTheMatchingConceptNameTagObjectI
 		assertNotNull(conceptNameTag);
 		assertEquals(conceptNameTag, conceptService.getConceptNameTagByName(conceptNameTag.getTag()));
 	}
+
+	/**
+	 * @see ConceptServiceImpl#saveConceptProposal(ConceptProposal)
+	 */
+	@Test
+	public void saveConceptReferenceRange_shouldReturnSavedConceptReferenceRangeObject() {
+		final Double HIGH_ABSOLUTE = 120.0;
+		
+		ConceptReferenceRange conceptReferenceRange = new ConceptReferenceRange();
+		conceptReferenceRange.setHiAbsolute(HIGH_ABSOLUTE);
+		conceptReferenceRange.setConceptNumeric(createConceptNumeric());
+		conceptReferenceRange.setCriteria(TEST_CRITERIA);
+		ConceptReferenceRange savedConceptReferenceRange = conceptService.saveConceptReferenceRange(conceptReferenceRange);
+		assertEquals(HIGH_ABSOLUTE, savedConceptReferenceRange.getHiAbsolute());
+
+		// Update
+		final Double NEW_HIGH_ABSOLUTE = 200.0;
+		savedConceptReferenceRange.setHiAbsolute(NEW_HIGH_ABSOLUTE);
+
+		ConceptReferenceRange updatedConceptReferenceRange = conceptService.saveConceptReferenceRange(savedConceptReferenceRange);
+		assertEquals(NEW_HIGH_ABSOLUTE, updatedConceptReferenceRange.getHiAbsolute());
+	}
+
+	/**
+	 * @see ConceptServiceImpl#saveConceptProposal(ConceptProposal)
+	 */
+	@Test
+	public void saveConceptReferenceRange_shouldFailGivenNull() {
+		assertThrows(IllegalArgumentException.class, () -> conceptService.saveConceptReferenceRange(null));
+	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptReferenceRangesByConceptId(Integer) 
+	 */
+	@Test
+	public void getConceptReferenceRangesByConceptId_shouldReturnConceptReferenceRangeIfFound() {
+		ConceptNumeric conceptNumeric = createConceptNumeric();
+		
+		ConceptReferenceRange conceptReferenceRange = new ConceptReferenceRange();
+		conceptReferenceRange.setHiAbsolute(120.0);
+		conceptReferenceRange.setConceptNumeric(conceptNumeric);
+		conceptReferenceRange.setCriteria(TEST_CRITERIA);
+		conceptService.saveConceptReferenceRange(conceptReferenceRange);
+
+		List<ConceptReferenceRange> savedConceptReferenceRange = conceptService.getConceptReferenceRangesByConceptId(conceptNumeric.getId());
+		assertFalse(savedConceptReferenceRange.isEmpty());
```

Comment:
size() would be better than isEmpty(), just in case more than the expected count is returned